### PR TITLE
Use Search API instead of Random API for gif

### LINKED ISSUE #22: [feature] a less random gif option
Right now our `gif` is a'ight. It uses the Giphy's random API to get a gif that is _loosely_ related to the input.
Would be nice if we could have a way to get a gif that is more closely related to the input, still with some degree of randomness, so that we don't see the same gif over and over

I believe Giphy API has some other endpoints that give you a list of giphys ordered by relevance. We can probably utilize that and use our own RNG here.

### PR DESCRIPTION
**Pull request by BrandonWLee**
Closes #22 

So I just used the Search endpoint instead and randomly chose a result.
Do you guys think it's ok to replace the old implementation or would we rather add a separate command for this endpoint?

### CODE REVIEW

File: commands/gif.js [hunk 1]
```diff
@@ -1,19 +1,47 @@
...
 
 exports.run = async (client, message, args, level) => {
   const start = Date.now();
-  const tags = args.join(" ");
-  giphy.random({
-    "tag": tags,
+  const query = args.join(" ");
+  giphy.search({
+    "q": query,
+    "limit": 100,
```
Comment (mynameisivan) [inline commands/gif.js:14]: Alternatively: Could we use offset and determine the bucket earlier? Not sure how large the results being returned is (or if it affects performance/memory)

Comment (BrandonWLee) [reply commands/gif.js:14]: If we don't know how many total results there will be and we set the offset to a number too high, then no results will be returned. For example. 'ooga' returns 33 results. If we set the offset to 40, the results.data length is 0.
```
{
   "data":
   [
   ],
   "pagination":
   {
       "total_count": 8,
       "count": 0,
       "offset": 40
   },
   "meta":
   {
       "status": 200,
       "msg": "OK",
       "response_id": "z6iowljy8q5tvqz2s0mup20d4fkkbpc5tknb7alq"
   }
}
```

Comment (mynameisivan) [reply commands/gif.js:14]: Good catch. Nice catch on 35-36

Comment (BrandonWLee) [reply commands/gif.js:14]: They call me the clutchmeister

**Pull request merged**